electionleaflets/apps/leaflets/forms.py

```diff
@@ -63,26 +46,94 @@ class Meta:
         fields = ('reviewed', )
 
 
-class PeopleModelChoiceField(forms.ModelChoiceField):
-    def label_from_instance(self, obj):
-        if obj.current_party:
-            party_name = obj.current_party.party.party_name
-        else:
-            party_name = "Independent"
+class PeopleRadioWidget(forms.RadioSelect):
 
-        return "{0} ({1})".format(
-            obj.name,
-            party_name,
-        )
+    def create_option(self, name, value, label, selected, index, subindex=None,
+                      attrs=None):
+        if not label:
+            label = "Not Listed"
+        else:
+            label = u"{0} ({1})".format(
+                label["person"]["name"],
+                label["party"]["party_name"],
+            )
+        return super(PeopleRadioWidget, self).create_option(name, value, label, selected, index,
+                                                            subindex, attrs)
 
 
 class PeopleForm(forms.Form):
+    # A temporary top-X list of parties per register to show if the candidate
+    # isn't shown. We will eventually create this list based on real data.
+    HARDCODED_PARTIES = {
+        'gb': [
+            (52, 'Conservative and Unionist Party'),
+            (63, 'Green Party'),
+            (53, 'Labour Party'),
+            (90, 'Liberal Democrats'),
+            (102, 'Scottish National Party (SNP)'),
+            (7931, 'The Brexit Party'),
+            (77, 'Plaid Cymru - The Party of Wales'),
+        ],
+        'ni': [
+            (103, 'Alliance - Alliance Party of Northern Ireland'),
+            (70, 'Democratic Unionist Party - D.U.P.'),
+            (55, 'SDLP (Social Democratic & Labour Party)'),
+            (39, 'Sinn Féin'),
+            (83, 'Ulster Unionist Party'),
+        ]
+    }
+
     def __init__(self, *args, **kwargs):
         super(PeopleForm, self).__init__(*args, **kwargs)
-        if 'people' in kwargs['initial']:
+        signer = Signer()
+
+        if "postcode_results" in kwargs['initial']:
+
+            postcode_results = kwargs['initial']["postcode_results"].json()
+
+            # We have a response, parse each candidacy in to a set
+            # of unique people as people can stand in more than one ballot
+            # for a postcode
+            unique_people = OrderedDict()
+            for date in postcode_results["dates"]:
+                for ballot in date["ballots"]:
+                    for candidacy in ballot["candidates"]:
```

Comment:
Should we be filtering out cancelled ballots here?
Its a bit of an odd one in this situation because if a ballot is rescheduled the leaflets were probably all printed.. but then if you think about my other comment, the case of a countermanded poll would be _super_-confusing because you get 2 ballots in quick succession with almost identical candidate lists.

Reply:
Yep, maybe this is a post-GE improvement?